src/file_dialog.rs

```diff
@@ -383,7 +389,33 @@ impl FileDialog {
         }
 
         self.update_keybindings(ctx);
-        self.update_ui(ctx);
+        self.update_ui(ctx, None);
+
+        self
+    }
+
+    /// Do an [update](`Self::update`) with a custom right panel ui.
+    ///
+    /// Example use cases:
+    /// - Show custom information for a file (size, MIME type, etc.)
+    /// - Embed a preview, like a thumbnail for an image
+    /// - Add controls for custom open options, like open as read-only, etc.
+    ///
+    /// See [`active_entry`](Self::active_entry) to get the active directory entry
+    /// to show the information for.
+    ///
+    /// This function has no effect if the dialog state is currently not `DialogState::Open`.
+    pub fn update_with_custom_right_panel(
```

Comment:
It's a bit of a long name, but I think being descriptive is more important than being short.

Reply:
Yes, I agree with you. I would suggest `update_with_right_panel_ui`. But I am happy with both names.

Reply:
I renamed it to `update_with_right_panel_ui`.